src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,46 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+type Props = {
+  selectedTodo: Todo,
+  onCloseModal: () => void,
+};
+
+export const TodoModal: React.FC<Props> = ({ selectedTodo, onCloseModal }) => {
+  const {
+    id,
+    title,
+    completed,
+    userId,
+  } = selectedTodo;
+
+  const [user, setUser] = useState<User | null>(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const userFromServer = await getUser(userId);
+
+        setUser(userFromServer);
+      } catch (err) {
+        setUser({
+          id: 0,
+          name: 'Anonym',
+          email: 'unknown',
+          phone: 'unknown',
+        });
+      }
+    })();
+  }, []);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
+      {!user ? (
```

Comment:
it would be better to check `selectedTodo` instead of `user`
```suggestion
      {!selectedTodo ? (
```